packages/snaps-cli/src/webpack/plugins.ts

```diff
@@ -0,0 +1,408 @@
+
+  /**
+   * The spinner to use for logging.
+   */
+  readonly #spinner?: Ora;
+
+  constructor(resolver: SnapsBuiltInResolver | false, spinner?: Ora) {
+    this.#resolver = resolver;
+    this.#spinner = spinner;
+  }
+
+  /**
+   * Apply the plugin to the Webpack compiler.
+   *
+   * @param compiler - The Webpack compiler.
+   */
+  apply(compiler: Compiler) {
+    compiler.hooks.afterCompile.tap(this.constructor.name, () => {
+      if (!this.#resolver) {
+        return;
+      }
+
+      const { unresolvedModules } = this.#resolver;
+      if (unresolvedModules.size === 0) {
+        return;
+      }
+
+      const formattedModules = new Array(...unresolvedModules)
+        .map((name) => indent(`• ${name}`, 2))
+        .join('\n');
+
+      warn(
+        `The snap attempted to use one or more Node.js builtins, but no browser fallback has been provided.\n` +
+          `The MetaMask Snaps CLI does not support Node.js builtins by default. If you want to use this module, you must provide a fallback: https://webpack.js.org/configuration/resolve/#resolvefallback.\n` +
+          `To disable this warning, set ${yellow(
+            '`plugins.builtInResolver`',
+          )} to ${yellow(
+            '`false`',
+          )} in your snap config file, or add the module to the ${yellow(
+            '`plugins.builtInResolver.ignore`',
+          )} array.\n\n${formattedModules}\n`,
+        this.#spinner,
+      );
+    });
+  }
+}
+
+/**
+ * The options for the {@link SnapsBundleWarningsPlugin}.
+ */
+export type SnapsBundleWarningsPluginOptions = {
+  /**
+   * Whether to show warnings if the `Buffer` global is used, but not provided
+   * by Webpack's `DefinePlugin`.
+   */
+  buffer?: boolean;
+};
+
+/**
+ * A plugin that logs a message when a snap uses the `Buffer` global. The
+ * MetaMask Snaps CLI does not support the `Buffer` global by default, and this
+ * plugin is used to warn the user when they try to use the `Buffer` global.
+ */
+export class SnapsBundleWarningsPlugin implements WebpackPluginInstance {
+  /**
+   * The spinner to use for logging.
+   */
+  readonly #spinner?: Ora;
+
+  /**
+   * The options for the plugin.
+   */
+  readonly #options: SnapsBundleWarningsPluginOptions;
+
+  constructor(
+    options: SnapsBundleWarningsPluginOptions = {
+      buffer: true,
+    },
+    spinner?: Ora,
+  ) {
+    this.#options = options;
+    this.#spinner = spinner;
+  }
+
+  /**
+   * Apply the plugin to the Webpack compiler.
+   *
+   * @param compiler - The Webpack compiler.
+   */
+  apply(compiler: Compiler) {
+    if (this.#options.buffer) {
+      this.#checkBuffer(compiler);
+    }
+  }
+
+  /**
+   * Check if the `Buffer` global is used, but not provided by Webpack's
+   * `DefinePlugin`.
+   *
+   * @param compiler - The Webpack compiler.
+   */
+  #checkBuffer(compiler: Compiler) {
+    const plugin = compiler.options.plugins?.find(
+      (instance) => instance instanceof ProvidePlugin,
```

Comment:
Couldn't `instanceof` fail if a user has multiple versions of Webpack installed?

Reply:
Maybe? I'm not sure how it works.

Reply:
I'm pretty sure that might be the case. Is there another way to detect the plugin?